pkg/ccl/backupccl/backup_test.go

```diff
@@ -3464,6 +3464,48 @@ func TestConcurrentBackupRestores(t *testing.T) {
 	}
 }
 
+func TestBackupWithMismatchedLocalityLayersFails(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	const numAccounts = 1
+	ctx := context.Background()
+	_, sqlDB, _, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, InitManualReplication)
+	defer cleanupFn()
+
+	const msg = "Mismatched backup layers are not supported."
+
+	_, err := sqlDB.DB.ExecContext(
+		ctx,
+		`BACKUP INTO (
+'nodelocal://0/default?COCKROACH_LOCALITY=default',
```

Comment:
since this isn't getting backported and because we want to move to using data driven tests whenever possible, could you make this driven :D